### PR TITLE
UBER-690 Intercom integration

### PR DESCRIPTION
**Pull request by aonnikov**
# Contribution checklist

## Brief description

Added basic Intercom integration.
1. Load and show Intercom widget on demand, it is available from toolbar and Help & Support page
2. Support platform localization
3. Shows notification icon on new message (only when widget is loaded)

TODO:
- [x] Show unread messages without loading Intercom widget
- [ ] Remove user hash generation from frontend
- [ ] Update tokens to Uberflow Intercom account
- [ ] Update Kubernetes deployment with env variables

Screenshot 2023-08-10 at 12 42 46

## Checklist

* [ ] - UI test added to added/changed functionality?
* [x] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [x] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: dev/docker-compose.yaml [hunk 1]
```diff
@@ -168,6 +171,20 @@ services:
   #     - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
   #   command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd
   #   restart: unless-stopped
+  # intercom:
```
Comment (BykhovDenis) [inline dev/docker-compose.yaml]: I'm not sure that we need it to local build

File: models/support/src/migration.ts
Comment (BykhovDenis) [inline models/support/src/migration.ts:1]: It I not necessary

File: models/support/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export { supportId } from '@hcengineering/support'
+export { supportOperation } from './migration'
+export { support as default }
+
+export const DOMAIN_SUPPORT = 'support' as Domain
+
+@Model(support.class.SupportConversation, preference.class.Preference)
+export class TSupportConversation extends TPreference implements SupportConversation {
+  @Index(IndexKind.Indexed)
+    account!: Ref<Account>
```
Comment (BykhovDenis) [inline models/support/src/index.ts]: We will have same value in createdBy and space.

File: dev/docker-compose.yaml (hunk 1 above)
Comment (aonnikov) [reply dev/docker-compose.yaml]: Yes, will be removed and updated Readme with instruction how to use it for testing.

File: models/support/src/index.ts (hunk 2 above)
Comment (aonnikov) [reply models/support/src/index.ts]: Will replace with "space"

File: models/support/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import preference, { TPreference } from '@hcengineering/model-preference'
+import support, { SupportConversation, SupportSystem, SupportWidgetFactory } from '@hcengineering/support'
+
+import { TDoc } from '@hcengineering/model-core'
+import { Resource } from '@hcengineering/platform'
+
+export { supportId } from '@hcengineering/support'
+export { support as default }
+
+export const DOMAIN_SUPPORT = 'support' as Domain
```
Comment (BykhovDenis) [inline models/support/src/index.ts:27]: We don't need it

**Review by BykhovDenis — APPROVED** [review]

File: plugins/support-resources/src/support.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    this.config = {
+      account: getCurrentAccount(),
+      workspace: get(location).path[1],
+      language: get(themeStore).language
+    }
+    this.updateWidgetConfig(this.config)
+
+    if (onStatusChanged !== undefined) {
+      const query = createQuery(true)
```
Comment (haiodo) [inline plugins/support-resources/src/support.ts:56]:
```suggestion
      this.query = createQuery(true)
```

File: plugins/workbench-resources/src/components/Workbench.svelte [hunk 5]
```diff
@@ -575,6 +576,18 @@
...
+  let supportStatus: SupportStatus | undefined = undefined
+  function handleSupportStatusChanged (status: SupportStatus) {
+    supportStatus = status
+  }
+
+  const supportClient = getResource(support.function.GetSupport).then((res) =>
+    res((status) => handleSupportStatusChanged(status))
+  )
+  onDestroy(async () => {
+    await supportClient?.then((support) => support.destroy())
```
Comment (haiodo) [inline plugins/workbench-resources/src/components/Workbench.svelte:588]: Why not simply form?
```suggestion
    (await supportClient?)?.destroy()
```

File: models/support/src/index.ts (hunk 3 above)
Comment (aonnikov) [reply models/support/src/index.ts:27]: It is here by mistake, forgot to remove

File: plugins/workbench-resources/src/components/Workbench.svelte (hunk 5 above)
Comment (aonnikov) [reply plugins/workbench-resources/src/components/Workbench.svelte:588]: Fixed

File: plugins/support-resources/src/support.ts (hunk 4 above)
Comment (aonnikov) [reply plugins/support-resources/src/support.ts:56]: Thanks, fixed

**Pull request merged**